gdi/gdi.c

```diff
@@ -71,6 +71,15 @@ struct dib_driver
     DWORD       padding;
 };
 static struct list dib_drivers = LIST_INIT(dib_drivers);
+
+static COLORREF check_colorref(COLORREF color)
+{
+    int type = color >> 16;
+    if ((type != 0x10ff) && ((type & 0xf00) != 0x100) && ((type & 0xf00) != 0x200))
```

Comment:
It does not work when MSbyte is 0x11, 0x21, 0x31...
```suggestion
    if ((type & 0xff00) != 0x0100 && (type & 0xff00) != 0x0200)
```
I think this is better.

Reply:
Are those valid though?  In https://github.com/wine-mirror/wine/blob/master/dlls/winex11.drv/palette.c#L816 it looks like 0x0100xxxx (windows seems to ignore the 3rd byte) is PALETTEINDEX, 0x10ffxxxx is DIBINDEX and 0x2xxxxxx is PALETTERGB otherwise the high byte should be 0 (or ignored).

Reply:
Experimentation shows this

`           win10  win31`
`00ffffff - white  white`
`10ffffff - black  white`
`100000ff - red    red`
`f0ffffff - black  white`
`e0ffffff - white  white`
`0fffffff - black  white`
`01ffffff - black  black`
`010000ff - black  black`
`02ffffff - white  white`
`020000ff - red    red`
`11ffffff - black  white`
`21ffffff - black  white`
`01000005 - purple purple`
`0f00000f - yellow black`
`ff00000f - yellow black`

which means DIBINDEX and PALETTERGB are ignored by win31 and while win10 assumes PALETTEINDEX if bit 24 is set win31 does if and only if bit 24 is set in the high byte.

Reply:
I did not consider DIBINDEX.
In Windows 3.1, DIBINDEX() is valid for DC which is created by DIB.DRV.